ddtrace/tracer.py

```diff
@@ -80,19 +81,131 @@ def configure(self, enabled=None, hostname=None, port=None, sampler=None, contex
         if context_provider is not None:
             self._context_provider = context_provider
 
-    def trace(self, name, service=None, resource=None, span_type=None, ctx=None, span_parent=None):
+    def start_span(self, name, service=None, resource=None, span_type=None):
         """
-        Return a span that will trace an operation called `name`. The context that generated
-        the Span may be provided, as well as the current parent Span.
+        Starts and returns a new ``Span`` representing a unit of work.
 
-        :param str name: the name of the operation being traced
+        :param str name: the name of the operation being traced.
+        :param str service: the name of the service being traced.
+        :param str resource: an optional name of the resource being tracked.
+        :param str span_type: an optional operation type.
+
+        To start a new root span::
+
+            >>> span = tracer.start_span("web.request")
+        """
+        span = Span(
+            self,
+            name,
+            service=service,
+            resource=resource,
+            span_type=span_type,
+        )
+        self.sampler.sample(span)
+
+        # add common tags
+        if self.tags:
+            span.set_tags(self.tags)
+
+        # create a new context for the root span
+        context = Context()
+        context.add_span(span)
+        return span
+
+    def start_child_span(self, name, parent_span, service=None, resource=None, span_type=None):
+        """
+        Starts and returns a new child ``Span`` from the given parent, representing a unit of work.
+
+        :param str name: the name of the operation being traced.
+        :param object parent_span: the parent span that creates this child.
+        :param str service: the name of the service being traced. If not set,
+                            it will inherit the service from its parent.
+        :param str resource: an optional name of the resource being tracked.
+        :param str span_type: an optional operation type.
+
+        To start a new child span::
+
+            >>> parent_span = tracer.start_span("web.request")
+            >>> span = tracer.start_child_span("web.worker", parent_span)
+        """
+        span = Span(
+            self,
+            name,
+            service=(service or parent_span.service),
+            resource=resource,
+            span_type=span_type,
+            trace_id=parent_span.trace_id,
+            parent_id=parent_span.span_id,
+        )
+        span._parent = parent_span
+        span.sampled = parent_span.sampled
+
+        # add common tags
+        if self.tags:
+            span.set_tags(self.tags)
+
+        # get the context from the parent span
+        parent_span.context.add_span(span)
+        return span
+
+    def start_child_from_context(self, name, context, service=None, resource=None, span_type=None):
```

Comment:
I think that this method is a little redundant, also error prone (``child_from_context`` but it could also be a root span if the ``Context`` is empty). I was following some thought about that just to see where it leads. But I think that the best way is to have:
```
def start_child_span(self, name, parent_span, service=None, resource=None, span_type=None):
    # parent_span can also be a Context; we will duck type it
```
or
```
def start_child_span(self, name, parent_span, service=None, resource=None, span_type=None, context=None):
    # explicit but it's weird that we can start a child from one parent, while passing a context that may
    # include another trace
```

Probably it's better following an approach [similar to OT](https://github.com/opentracing/opentracing-python/blob/master/opentracing/tracer.py#L72-L74)?